### DISCUSSION TITLE
Implement early circuit breaking for 3rd party services

### DISCUSSION THREAD

**Discussion by 0-vortex** [open-sauced/open-sauced#1083] [Ideas]
Not the only one of it's kind but something I wanted to try for a while: [nodeshift/opossum](https://nodeshift.dev/opossum/)

The use case for this would be GitHub API/GQL and OneGraph. Some interesting reads portraying the concepts:
- https://www.bennadel.com/blog/3299-building-a-circuit-breaker-for-node-js.htm
- https://blog.appsignal.com/2020/07/22/nodejs-resiliency-concepts-the-circuit-breaker.html
- https://blog.bearer.sh/build-a-circuit-breaker-in-node-js/

Some people may have come in contact with such ideas through [axios interceptors](https://axios-http.com/docs/interceptors) or [crawler interceptors](https://www.checklyhq.com/learn/headless/)

**Comment by mtfoley** [comment]
I think if I wrote my own I'd call it "band gap" but only because of the circuit breaker thing :-)
